packages/admin/cms-admin/src/preview/site/SitePreview.tsx

```diff
@@ -113,13 +116,26 @@ function SitePreview({ resolvePath, logo = <CometColor sx={{ fontSize: 32 }} />
         }
     });
 
-    const initialPageUrl = `${siteConfig.sitePreviewApiUrl}?${new URLSearchParams({
-        scope: JSON.stringify(scope),
-        path: iframePath,
-        settings: JSON.stringify({
-            includeInvisible: showOnlyVisible ? false : true,
-        }),
-    }).toString()}`;
+    const { data, error, refetch } = useQuery<GQLSitePreviewJwtQuery>(
+        gql`
+            query SitePreviewJwt($scope: JSONObject!, $path: String!, $includeInvisible: Boolean!) {
```

Comment:
One downside to a mutation would be that it can't be queried like this here. But we could create it before even navigating to the site preview.